docs/05-infrastructure-nodes.md

```diff
@@ -0,0 +1,318 @@
+    sigs.k8s.io/cluster-api-machine-type: worker
+  name: 190125-3-worker-us-west-1b
+  namespace: openshift-cluster-api
+  resourceVersion: "9027"
+  selfLink: /apis/cluster.k8s.io/v1alpha1/namespaces/openshift-cluster-api/machinesets/190125-3-worker-us-west-1b
+  uid: 591b4d06-20ba-11e9-a880-068acb199400
+```
+
+### Selector
+The `MachineSet` defines how to create `Machines`, and the `Selector` tells
+the operator which machines are associated with the set:
+
+```YAML
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      sigs.k8s.io/cluster-api-cluster: 190125-3
+      sigs.k8s.io/cluster-api-machineset: 190125-3-worker-us-west-1b
+```
+
+In this case, the cluster name/ID is `190125-3` and there is an additional
+label for the whole set.
+
+### Template Metadata
+The `template` is the part of the `MachineSet` that templates out the
+`Machine`. The `template` itself can have metadata associated, and we need to
+make sure that things match here when we make changes:
+
+```YAML
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        sigs.k8s.io/cluster-api-cluster: 190125-3
+        sigs.k8s.io/cluster-api-machine-role: worker
+        sigs.k8s.io/cluster-api-machine-type: worker
+        sigs.k8s.io/cluster-api-machineset: 190125-3-worker-us-west-1b
+```
+
+### Template Spec
+The `template` needs to `spec`ify how the `Machine`/node should be created. You will notice that the `spec` and, more specifically, the `providerSpec` contains all of the important AWS data to help get the `Machine` created correctly and bootstrapped.
+
+In our case, we want to ensure that the resulting node inherits one or more specific labels. As you've seen in the examples above, labels go in `metadata` sections:
+
+```YAML
+  spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          ami:
+            id: ami-08871aee06d13e584
+....
+```
+
+By default the `MachineSets` that the installer creates do not apply any
+additional labels to the node.
+
+## Defining a Custom MachineSet
+Now that you've analyzed an existing `MachineSet` it's time to go over the
+rules for creating one, at least for a simple change like we're making:
+
+1. Don't change anything in the `providerSpec`
+1. Don't change any instances of `sigs.k8s.io/cluster-api-cluster: <clusterid>`
+1. Give your `MachineSet` a unique `name`
+1. Make sure any instances of `sigs.k8s.io/cluster-api-machineset` match the `name`
+1. Add labels you want on the nodes to `.spec.template.spec.metadata.labels`
+1. Even though you're changing `MachineSet` `name` references, be sure not to change the `subnet`.
+
+This sounds complicated, so let's go through an example. Go ahead and dump
+one of your existing `MachineSets` to a file, and then open it with your
+favorite text editor. For example:
+
+    oc get machineset 190125-3-worker-us-west-1b -o yaml -n openshift-cluster-api > infra-machineset.yaml
+
+### Clean It
```

Comment:
`oc get <object> --export` gives you scrubbed objects.

Reply:
@abhinavdahiya so are you saying we should use export instead of get here?

Reply:
I just dug into `--export` and didn't find it particularly helpful. It ended up blanking out some things that were useful (eg: `namespace`) and so I'd prefer to leave it as-is with vanilla `get`.